src/transactions/TransactionFrame.cpp

```diff
@@ -1790,6 +1800,9 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
             }
         }
 
+        // we push the diagnostic events no matter what
+        outerMeta.pushDiagnosticEvents(std::move(txDiagnostics));
```

Comment:
This is causing a meta hash mismatch in the tests. Previously, soroban meta was only set for soroban transactions, but this will activate `SorobanTransactionMeta` when we previously didn't.

Reply:
fixed. thanks for the catch!